src/dmd/root/ctfloat.d

```diff
@@ -167,26 +167,43 @@ extern (C++) struct CTFloat
     }
 
     @system
-    static real_t parse(const(char)* literal, bool* isOutOfRange = null)
+    extern (D) private static real_t parse(alias parseFn)(const(char)* literal, bool* isOutOfRange)
     {
         errno = 0;
         version(CRuntime_DigitalMars)
         {
             auto save = __locale_decpoint;
             __locale_decpoint = ".";
+            scope(exit) __locale_decpoint = save;
```

Comment:
Leave the old way, as this way causes an exception handling frame to be built, which is slow.

Reply:
`parseFn` should be nothrow in all cases, hence no `finally` should be necessary.

Reply:
The operative word is "should be". Check and see.

Reply:
Right; added `nothrow` to the `parse()` function.